Add an events guide

choldgraf opened the pull request:
This PR does a few things:

- Adds a section to the docs for "usecase guides". I imagine this is a place where we can put information that is useful for particular use-cases like "Pangeo Style" workflows, "teaching", "events", etc.
- Adds a short "events guide" with a few tips for running events on a hub. Over time we can update this with more best-practices as we learn them.
- Re-organizes the pages slightly so that the left sidebar is less cluttered, and nests the `network` page under a `topics` folder rather than being a top-level page.
- Updates the environment information after feedback from PaleoHackWeek

If nobody objects, I'd like to get this merged in and see if it is useful for https://github.com/2i2c-org/infrastructure/issues/810 and we can iterate on this content further when debriefing after the OpenScapes hub

closes #111

GeorgianaElena commented on admin/howto/environment.md:
```diff
@@ -5,7 +5,92 @@ configured environment with base libraries, user interfaces and
+(environment/custom)=
+## Customize your user environment
+
+Whie all hubs [come with a default environment](environment/default), it is possible to create a custom user environment for the hub.
```
```suggestion
While all hubs [come with a default environment](environment/default), it is possible to create a custom user environment for the hub.
```

GeorgianaElena reviewed (approved):
LGTM :rocket:

sgibson91 commented on admin/howto/environment.md:
````diff
@@ -5,7 +5,92 @@ configured environment with base libraries, user interfaces and
+you distribute, so when run it'll install necessary packages. For R,
+you can use `install.packages("package-name")` as you normally would.
+
+```{warning}
````
What's the difference between the `%` and `!` prefixes in the 2 `pip` examples? Do they cause the packages to be installed in different places, or is that solely the `--user` flag?

(This is probably more for my own understanding than a requirement for the docs?)

sgibson91 reviewed (approved):
LGTM!

damianavila commented on admin/howto/environment.md:
```diff
@@ -5,7 +5,92 @@ configured environment with base libraries, user interfaces and
+You can temporarily install packages in your environment that will
+just last the duration of your user session. They will get wiped out
+when your user server is stopped, to ensure that you always start from
+a 'clean slate' environment.
```
```suggestion
a 'clean state' environment.
```

Maybe?

sgibson91 replied:
What Chris has written is correct: https://www.dictionary.com/browse/clean--slate BUT it's a good idea to reduce the use English idioms in our docs as much as possible

sgibson91 replied:
Maybe "always start from the default environment"? Or "provided environment"?

damianavila reviewed (approved):
Minor comment, but LGTM.
Let's iterate on this one!

choldgraf commented:
OK I believe that I've addressed all of the PR comments, thanks everybody for the helpful feedback! And agreed w/ @damianavila we should keep iterating on this!
